File: frontend/lib/features/pet/domain/repositories/pet_repository.dart
import 'package:dartz/dartz.dart'; import 'package:myguard_frontend/core/error/failures.dart'; import 'package:myguard_frontend/core/network/paginated_response_model.dart'; import 'package:myguard_frontend/features/pet/domain/entities/pet_entity.dart';

abstract class PetRepository {
  Future<Either<Failure, PaginatedResponseModel<PetEntity>>> getPets({int page, int size, String? societyId});
  Future<Either<Failure, PetEntity>> getPetById(String id);
  Future<Either<Failure, PetEntity>> registerPet(Map<String, dynamic> data);
  Future<Either<Failure, PetEntity>> updatePet(String id, Map<String, dynamic> data);
  Future<Either<Failure, void>> deletePet(String id);
  Future<Either<Failure, VaccinationEntity>> addVaccination(String petId, Map<String, dynamic> data);
  Future<Either<Failure, List<VaccinationEntity>>> getVaccinations(String petId);
}

File: backend/src/main/java/com/myguard/pet/repository/PetRepository.java
package com.myguard.pet.repository;

import com.myguard.pet.view.PetEntity;
import com.myguard.pet.view.VaccinationEntity;

import java.util.List;
import java.util.Optional;

public interface PetRepository {

    PetEntity savePet(PetEntity pet);
    Optional<PetEntity> findPetById(String id);
    PetEntity updatePet(PetEntity pet);
    void deletePet(String id);
    List<PetEntity> findPets(String societyId, int page, int size);
    long countPets(String societyId);

    VaccinationEntity saveVaccination(VaccinationEntity vaccination);
    List<VaccinationEntity> findVaccinationsByPetId(String petId);
}
